docs/source/en/tasks/object_detection.mdx

```diff
@@ -0,0 +1,487 @@
+4. Call `train()`` to finetune your model.
+
+When loading the model from the same checkpoint that you used for the preprocessing, remember to pass the label to
+id and id to label maps that you created earlier from the dataset's metadata.
+
+```py
+>>> from transformers import DetrForObjectDetection
+
+>>> model = DetrForObjectDetection.from_pretrained(
+...     checkpoint,
+...     id2label=id2label,
+...     label2id=label2id,
+...     ignore_mismatched_sizes=True,
+... )
+```
+
+In the `TrainingArguments` use `output_dir` to specify where to save your model, then configure hyperparameters as you see fit.
+An important note: do not remove unused columns because this will drop the image column. Without the image column, you
+can't create `pixel_values`. For this reason, set `remove_unused_columns` to `False`.
+If you wish to share your model by pushing to the Hub, set `push_to_hub` to `True` (you must be signed in to Hugging
+Face to upload your model).
+
+```py
+>>> from transformers import TrainingArguments
+
+>>> training_args = TrainingArguments(
+...     output_dir="detr-resnet-50_fine_tuned_cppe5",
+...     per_device_train_batch_size=8,
+...     num_train_epochs=10,
+...     fp16=False,
+...     save_steps=200,
+...     logging_steps=50,
+...     learning_rate=1e-5,
+...     save_total_limit=2,
+...     remove_unused_columns=False,
+...     push_to_hub=True,
+... )
+```
+
+Finally, bring everything together, and call `train`:
+
+```py
+>>> from transformers import Trainer
+
+>>> trainer = Trainer(
+...     model=model,
+...     args=training_args,
+...     data_collator=collate_fn,
+...     train_dataset=cppe5["train"],
+...     tokenizer=image_processor,
+... )
+
+>>> rainer.train()
+```
+
+## Evaluate
+Object detection models are commonly evaluated with a set of [COCO-style metrics](https://cocodataset.org/#detection-eval).
+You can use one of the existing metrics implementations, here you'll use the one from `torchvision`.
+
+Save your trained model to use for evaluation. Alternatively, you can evaluate the model that you have pushed to the Hub.
+
+```py
+>>> trainer.save_model("./detr-resnet-50_fine_tuned_cppe5")
+```
+
+To use the `torchvision` evaluator, you'll need to prepare a ground truth COCO dataset. The API to build a COCO dataset
+requires the data to be stored in a certain format, so you'll need to save images and annotations to disk first. Just like
+when you prepared your data for training, the annotations from the `cppe5["test"]` need to be formatted. However, images
+should stay as they are.
+
+The following example shows how you can:
+- Create a torchvision CocoDetection dataset
+- Reformat annotations from the `cppe5["test"]`
+- Save images and annotations into files that you can create an instance of CocoDetection dataset from
+- Use `cocoevaluate` metric to compute the [COCO metrics](https://cocodataset.org/#detection-eval).
+
+```py
```

Comment:
This code snippet is a bit long. 

I would break it down into pieces and places connectors in between them. 

* Preparing `CocoDetection` dataset for compatibility.
* Serializing images. 
* Evaluation. 

Also consider a different name than `CocoDetection` since it might confuse the readers. We aren't actually evaluating on COCO here.

Reply:
This is indeed way too long for this guide. It should be moved to a more advanced guide or juse be part in an example (this whole guide would make a great example once converted to a Python script) that we can link here.

Reply:
Phew, the evaluation part is long indeed. Unfortunately, there's no simpler way (yet) to properly use the coco metrics. I can split this into several pieces and rename parts so that it isn't as confusing.

Reply:
Yeah let's start with that.

Reply:
Split into parts and added some explanation between the blocks (in 1a38260b27ef977e0d82ba1c918eeb19c8f3dcb7), however, I may need to do another pass at the evaluation part. @younesbelkada is working on optimizing this.